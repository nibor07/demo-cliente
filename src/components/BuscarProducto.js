import React, { useState } from 'react'
import PropTypes from 'prop-types';
import lidercategoria from '../img/lider-categoria.jpg'; 
import shop from '../img/compra.jpg';

  const styleLogos = {
    textAlign: "center",
    height: "35px",
    marginTop: "10px",
    zIndex: 6
  };

  export const BuscarProducto = ( { setProductos } ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) =>{
        e.preventDefault();

        if( inputValue.trim().length > 0 ){
            setProductos(  prods => [ inputValue ]);
            setInputValue('');
        }
        
    }

    return (
        <div className='container'> 
        <div className="row row-cols-4" style={styleLogos}>
            <div className="col"><img src={lidercategoria} alt="Seleccione las categorías" /> </div>
            <div className="col-6">
                <form onSubmit={ handleSubmit }>
                    <div>
                        <input
                            name="search"
                            value={ inputValue }
                            className="input is-rounded"
                            type="text"
                            placeholder="¿Qué estás buscando?"
                            onChange={handleInputChange}
                        />
                    </div>
                </form>
            </div>
            <div className="col" align="right"><img src={shop} alt="Pagar"/></div>
        </div>
        </div>
    )
}

BuscarProducto.propTypes = {
    setProductos: PropTypes.func.isRequired
}