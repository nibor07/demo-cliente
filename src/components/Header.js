import React from 'react';
import agenda from '../img/agenda_icon.svg';


const style = {
    textAlign: "left",
    fontSize: "8px",
    backgroundColor: "#041e42",
    color: "#fff",
    position: "relative",
    zIndex: 4,
    height: "30px",
    width: "100%"   
  };

  const styleAgendaLocal = {
    color: "#fff",
    fontSize: "11px"
  };

const Header = () => {
    return(
        <div className="container-full">
            <div className="row rows-cols-8" style={style}>
                <div className= "col-8">
                    <span className="header" background-color="#041e42">Servicio al cliente de Lider.cl: WhatsApp 
                        <a href="https://api.whatsapp.com/send?phone=56957211492&amp;text=Hola!%20Tengo%20una%20consulta">+56957211492
                        </a>
                        | Horario: de lunes a viernes de 8:00 a 22:00 y s√°bado, domingo y festivos de 9:00 a 22:00
                    </span>
                </div>
                <div className="col">
                <span  background-color="#041e42">
                    <a href="https://www.lider.cl/agenda/#JuntosNosCuidamos" className="styled__StoreLink-sc-17jbrkg-0 kkNTwS">
                        <img height="13px" alt="agenda" src={agenda}/> &nbsp;
                        <span style={styleAgendaLocal}>agenda tu visita</span>
                    </a>
                    </span>
                </div>
                <div className="col-full">
                    <span className="col" background-color="#041e42">
                    <a href="https://www.lider.cl/tiendas" className="styled__StoreLink-sc-17jbrkg-0 kkNTwS">
                    <img height="13px" alt="locales" src={agenda}/>&nbsp;
                        <span style={styleAgendaLocal}>locales y horarios</span>
                    </a>
                    </span>
                </div>

            </div>
        </div>
    );
}

export default Header;