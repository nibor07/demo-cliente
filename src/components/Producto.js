import React from 'react'

export const Producto = ({ producto }) => {

    let imagen = "https://"+producto.url;

    const styleMarca = {
        fontWeight: "bold",
        color: "rgb(0, 0, 0)"
    };
    
    return (
        <div className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4">
            <div className="card">
                <img src={imagen} alt={imagen} className="card-img-top" />
            {    producto.isDescuento ? 
                <div className="card-body">
                    <p className="card-text"><span style={styleMarca}>{producto.brand}</span> <span>{producto.description}</span></p>
                    
                    <p className="card-text precio">{producto.precioConDescuento} <span className="porcentaje-descuento">&nbsp; 50%</span></p>
                    <p className="card-text precio-referencia">{producto.price} Vistas</p>
                </div>
                :
                <div className="card-body">
                    <p className="card-text"><span style={styleMarca}>{producto.brand}</span> <span>{producto.description}</span></p>
                    <p className="card-text precio">{producto.price} Vistas</p>
                </div>
            }
                <div className="card-footer">
                    <a
                        href={imagen}
                        target="_blank"
                        rel="noopener noreferrer"
                        className="btn btn-outline-primary"
                    >Agregar</a>
                </div>
            </div>
        </div>
    );
}