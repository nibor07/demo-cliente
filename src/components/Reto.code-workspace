{
	"folders": [
		{
			"path": "..\\..\\..\\..\\..\\Walmart\\MERN-Calendar-Backend-1.0.0"
		},
		{
			"path": "..\\..\\..\\..\\demo-server"
		},
		{
			"path": "..\\.."
		},
		{
			"path": "..\\..\\..\\..\\react_pixabay-master"
		},
		{
			"path": "..\\..\\..\\..\\react-hero-app-0.16.0"
		},
		{
			"path": "..\\..\\..\\..\\test-mocha-chai-1.0.0"
		},
		{
			"path": "..\\..\\..\\..\\express-mongo-master"
		}
	],
	"settings": {}
}