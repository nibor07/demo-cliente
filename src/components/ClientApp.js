import React, { useState } from 'react'
import { BuscarProducto } from './BuscarProducto';

export const ClientApp = (seach) => {

    console.log( 'URL: ', process.env.REACT_APP_BACKEND_URL);
    return (
        <div>
            <BuscarProducto search='hola'/>
        </div>
    )
}