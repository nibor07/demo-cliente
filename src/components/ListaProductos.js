import React from 'react';
import { useFetchProductos } from '../hooks/useFetchProductos';
import { Producto } from './Producto';
import loSentimos from '../img/lo_sentimos.webp'; 

const styleLogo = {
    textAlign: "center",
    height: "100px"
  };

const styleResultado = {
    textAlign: "center",
    height: "35px"
};

  export const ListaProductos = ({ producto }) => {

    //let query = producto;
    const { data:productos } = useFetchProductos( producto );
    console.log('data=>', productos);


    if(productos){
        
        return (
            <div className="col-12 p-5 row">
            {
                productos.map( producto => (
                    <Producto
                        key={ producto.id }
                        producto={producto}
                    />
                ))}



        </div>
        );
    }else{
        return(
            <div>
            <div className="rows sin-resultado is-5">
                <div className="btn-centrado" ><img src={loSentimos} style={styleLogo} alt="Lo sentimos!"/></div>
                <div className="btn-centrado mensaje-resultado">¡Lo sentimos!</div>
                <div className="btn-centrado mensaje-sin-dato" >No encontramos resultados para tu búsqueda de <span>"{producto}" </span></div>
                <div className="btn-centrado linea"></div>
                <div className="btn-centrado cFwMqv" style={styleResultado}><spam className="tips">Algunos tips:</spam> Revisa si escribiste correctamente lo que buscas. Puedes probar nuevamente usando términos menos específicos.</div>
            </div>
            </div>
        )

    }
}