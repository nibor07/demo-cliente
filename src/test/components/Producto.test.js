import React from 'react';
import { shallow } from 'enzyme'
import { Producto } from '../../components/Producto';

describe('Tests de <Producto />', () => {
    
    const producto = {  id: 123412,
                        url: "wwww.lider.cl/1.jpg",
                        brand: "Televisor DAEWOO",
                        description: "jausau",
                        precioConDescuento: 23421,
                        isDescuento: true,
                        price: "123"                        
                    };
    
    const wrapper = shallow(<Producto key={ producto.id }
                                    producto={producto}
                        />);
    
    test('debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe tener un pÃ¡rrafo con el brand del producto buscado', () => {
        
        const hdos = wrapper;
        //TODO revisar esta pruebaw
        console.log('data==>',hdos);
        //expect( hdos.text().trim() ).toBe( producto.price );

    })

    *test('debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        console.log(img);
        expect( img.prop('src') ).toBe( "https://"+producto.url );
        expect( img.prop('alt') ).toBe( "https://"+producto.url );

    });

})
