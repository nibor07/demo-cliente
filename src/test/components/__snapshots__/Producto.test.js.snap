// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de <Producto /> debe mostrar el componente correctamente 1`] = `
<div
  className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4"
>
  <div
    className="card"
  >
    <img
      alt="https://wwww.lider.cl/1.jpg"
      className="card-img-top"
      src="https://wwww.lider.cl/1.jpg"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <span
          style={
            Object {
              "color": "rgb(0, 0, 0)",
              "fontWeight": "bold",
            }
          }
        >
          Televisor DAEWOO
        </span>
         
        <span>
          jausau
        </span>
      </p>
      <p
        className="card-text precio"
      >
        $23.421
         
        <span
          className="porcentaje-descuento"
        >
          Â  50%
        </span>
      </p>
      <p
        className="card-text precio-referencia"
      >
        $123
      </p>
    </div>
    <div
      className="card-footer"
    >
      <a
        className="btn btn-outline-primary"
        href="https://wwww.lider.cl/1.jpg"
        rel="noopener noreferrer"
        target="_blank"
      >
        Agregar
      </a>
    </div>
  </div>
</div>
`;
